jax/_src/dlpack.py

```diff
@@ -78,9 +85,17 @@ def to_dlpack(x: Array, take_ownership: bool = False,
     warnings.warn(
         "take_ownership in to_dlpack is deprecated and it is a no-op."
     )
-  return xla_client._xla.buffer_to_dlpack_managed_tensor(
-      x.addressable_data(0), stream=stream
-  )  # type: ignore
+
+  if max_version is None or max_version[0] >= DLPACK_VERSION[0]:
+      return _to_dlpack(x, stream)
+  elif max_version >= MIN_DLPACK_VERSION:
+    # Legacy path
+    pass
```

Comment:
It seems strange to return `None` in this case

Reply:
This branch is never hit and is really just a template for if/when we add a legacy path

Reply:
What is the legacy path? Is this something that users could conceivably hit? If so then we should do something like this:
```python
raise NotImplementedError("dlpack legacy path is not implemented")
```
If this branch should truly never be hit by any conceivable user code, then I'd do something like
```python
raise RuntimeError("This branch should be unreachable. Please open a bug if you see this.")
```

Reply:
Right now I've set both `DLPACK_VERSION` and `MIN_DLPACK_VERSION` to `(0, 1)`, so currently the path is inaccessible. When we add a way to produce the newer `DLManagedTensorVersioned` format in XLA, then we increment `DLPACK_VERSION` and this path will actually be usable and the _current_ methodology will become the legacy path -- i.e. `_to_dlpack() --> _legacy_to_dlpack()` and we we'll have a new `_to_dlpack()` to match the altered XLA DLPack producer.

In the interim I will add the `RuntimeError` you suggested so that its role is explicit :)